### PR TITLE
Encryption

### PR DESCRIPTION
**Pull request by timbertson**
Building on #683

### CODE REVIEW

File: config.properties.example [hunk 1]
```diff
@@ -173,6 +173,18 @@ kinesis_stream=maxwell
...
+
+# convert the data field to a json string and encrypt it
+#encrypt_data=true
+
+# convert the whole payload to a json string and encrypt it
+#encrypt_all=true
+
+# specify the encryption key and secret key
+#encryption_key=aaaaaaaaaaaaaaaa
+#secret_key=RandomInitVector
```
Comment (liulikun) [inline config.properties.example]: outdated here

File: docs/docs/config.md [hunk 2]
```diff
@@ -182,6 +182,10 @@ exclude_tables                 | PATTERN                             | ignore up
 blacklist_dbs                  | PATTERN                             | ignore updates AND schema changes from databases (see warnings below) |
 blacklist_tables               | PATTERN                             | ignore updates AND schema changes from tables named like PATTERN (see warnings below) |
 &nbsp;
+**encryption**
+encrypt_data                   | BOOLEAN                             | encrypt the data field of the payload               | false
+encrypt_all                    | BOOLEAN                             | encrypt the entire payload                          | false
```
Comment (liulikun) [inline docs/docs/config.md]: Another way to configure this: `encrypt=[data|all|none]`, default is `none`.

File: src/main/java/com/zendesk/maxwell/producer/MaxwellOutputConfig.java [hunk 3]
```diff
@@ -4,7 +4,14 @@
 import java.util.ArrayList;
 import java.util.regex.Pattern;
 
+
 public class MaxwellOutputConfig {
+	public enum Encryption {
```
Comment (liulikun) [inline src/main/java/com/zendesk/maxwell/producer/MaxwellOutputConfig.java]: `EncryptionMode`?

**Comment by liulikun** [comment]
Looks good!

File: docs/docs/config.md (hunk 2 above)
Comment (timbertson) [reply docs/docs/config.md]: Yeah that would probably be better, since you're not allowed to specify both anyway

**Comment by liulikun** [comment]
👍

**Pull request merged**